src/components/Chat/Chat.jsx

```diff
@@ -46,9 +71,182 @@ const Chat = ({
   const isChatVisible = context?.isChatVisible;
   const setIsChatVisible = context?.setIsChatVisible;
 
+  const markAsRead = async chatId => {
+    try {
+      if (messages.length > 0) {
+        const lastMessageId = messages[messages.length - 1]?.id;
+        if (lastMessageId) {
+          await axiosClient.patch(`/chats/${chatId}/messages/last-read`, {
+            lastReadMessageId: lastMessageId,
+          });
+        }
+      }
+    } catch (error) {
+      console.error('Error updating last read message:', error);
+    }
+  };
+
+  const fetchPublicMessages = async (pageNumber = 0) => {
+    setIsFetchingMore(true);
+    try {
+      const response = await axiosClient.get(`/chats/${id}/messages`, {
+        params: {
+          size: 20,
+          page: pageNumber,
+          sort: 'creationDate,desc',
+        },
+      });
+
+      const { content, page: pageData } = response.data;
+
+      setMessages(prevMessages => [...content, ...prevMessages]);
+      setPage(pageData.number + 1);
+      setHasMore(pageData.number + 1 < pageData.totalPages);
+      return content;
+    } catch (error) {
+      console.error('Error fetching messages:', error);
+    } finally {
+      setIsFetchingMore(false);
+    }
+  };
+
+  const fetchMessages = async (pageNumber = 0) => {
+    setIsFetchingMore(true);
+    try {
+      const response = await axiosClient.get(`/chats/${id}/messages/read`, {
+        params: {
+          size: 20,
+          page: pageNumber,
+          sort: 'creationDate,desc',
+        },
+      });
+
+      const { content, page: pageData } = response.data;
+
+      setMessages(prevMessages => [...content, ...prevMessages]);
+      setPage(pageData.number + 1);
+      setHasMore(pageData.number + 1 < pageData.totalPages);
+      return content;
+    } catch (error) {
+      console.error('Error fetching messages:', error);
+    } finally {
+      setIsFetchingMore(false);
+    }
+  };
+
+  const fetchUnreadMessages = async (pageNumber = 0) => {
+    try {
+      const response = await axiosClient.get(`/chats/${id}/messages/unread`, {
```

Comment:
The same? and maybe we should move all this functions to separate file in order to clean this component.

Reply:
done
yes, I thought about it